Source/UI/NewUI/BillingForm/ViewModel/DefaultBillingFormViewModel.swift

```diff
@@ -214,11 +215,17 @@ extension DefaultBillingFormViewModel: BillingFormTextFieldDelegate {
 // MARK: - Billing form view controller Delegate
 
 extension DefaultBillingFormViewModel: BillingFormViewControllerDelegate {
+    func phoneNumberIsUpdated(number: String?) {
+        guard let number = number else { return }
+        let index = BillingFormCell.phoneNumber(nil).index
+        textValueOfCellType[index] = number
+        updatedRow = index
```

Comment:
I do not think this code reads at all. 

As a lifecycle update, I filter out valid updates.
I get the index from the variable of a type method assigned of a component unrelated to my purpose.
I then insert the number received into an array at the index given by said variable of uncertain origins.
Finally I keep a record of that index locally.

Just to be clear, I am not assessing the functionality of this, I trust since in a PR it works 100% as you need it to. I am only asking if there is any way this method could become legible and more logical. Personally looking at this I have no idea how it works or what's going on and why :)